notebook/notebookapp.py

```diff
@@ -1854,17 +1845,13 @@ def launch_browser(self):
         else:
             open_file = self.browser_open_file
 
-        if not self.use_redirect_file:
-            if browser:
-                b = lambda: browser.open(url_path_join(self.connection_url, uri),
-                                        new=self.webbrowser_open_new)
-                threading.Thread(target=b).start()
-
+        if self.use_redirect_file:
+            assembled_url = urljoin('file:', pathname2url(open_file))
         else:
-            b = lambda: browser.open(
-                urljoin('file:', pathname2url(open_file)),
-                new=self.webbrowser_open_new)
-            threading.Thread(target=b).start()
+            assembled_url = url_path_join(self.connection_url, uri)
+        
+        b = lambda: browser.open(assembled_url, new=self.webbrowser_open_new)                            
+        threading.Thread(target=b).start()
```

Comment:
Created an intermediate variable (`assembled_url`) that is handled depending on `use_redirect_file`. This is now sent to the same `browser.open` call, removing the boilerplate previously written.

Reply:
Also fixed the `if not/else` to be `if/else`